File: test/features/step_definitions/pupil-spa/loadingPageSteps.js
// Step definitions
/* global browser expect */

const {defineSupportCode} = require('cucumber')

defineSupportCode(function ({Given, When, Then}) {
  Then(/^I have read the instructions$/, function () {
    browser.get(this.config.spaUrl + 'sign-in')
    this.spaSignInPage.login('abc12345', '9999a')
    this.spaSignInSuccessPage.readInstructions.click()
    return this.spaStartPage.warmUpIntroLink.click()
  })

  Then(/^I choose to start the warm up questions$/, function () {
    return this.spaWarmUpIntroPage.startLink.click()
  })

  Then(/^I should have (\d+) seconds before I see the first question$/, function (int) {
    this.waitForVisibility(this.spaWarmUpQuestionPage.question, int * 1100)
    return expect(this.spaWarmUpQuestionPage.question.isPresent()).to.eventually.be.true
  })
})

File: pupil-spa/test/features/step_definitions/loading_page_steps.rb
Given(/^I have read the instructions$/) do
  step 'I have logged in'
  confirmation_page.read_instructions.click
end

When(/^I choose to start the warm up questions$/) do
  start_page.start_check.click
  warm_up_page.start_warm_up_questions.click
end

Then(/^I should have (\d+) seconds before I see the first question$/) do |arg1|
  check_page.wait_for_preload
  expect(check_page).to have_question

end
